ethers-providers/src/pubsub.rs

```diff
@@ -78,20 +78,17 @@ where
     P: PubsubClient,
     R: DeserializeOwned,
 {
-    type Item = R;
+    type Item = Result<R, serde_json::Error>;
 
     fn poll_next(self: Pin<&mut Self>, ctx: &mut Context) -> Poll<Option<Self::Item>> {
         if !self.loaded_elements.is_empty() {
-            let next_element = self.get_mut().loaded_elements.pop_front();
+            let next_element = self.get_mut().loaded_elements.pop_front().map(|s| Ok(s));
             return Poll::Ready(next_element)
         }
 
         let this = self.project();
         match futures_util::ready!(this.rx.poll_next(ctx)) {
-            Some(item) => match serde_json::from_str(item.get()) {
-                Ok(res) => Poll::Ready(Some(res)),
-                _ => Poll::Pending,
```

Comment:
I think the reasoning behind this is, that deserializing events (in most cases) works, because this is supposed to be used with correct bindings anyway so returning Result here is not really necessary (in most cases)

Reply:
I mean if it is the case that it silently stops after a single bad message, one could argue it'd be better to panic than to leave the program in a non functional state. And since this doesn't seem to be a normal issue, i would suspect it is exceedingly rare in normal operation... If it becomes a problem one could say to use (the theoretical) `TrySubscriptionStream` instead to avoid panics/catch errors.

Reply:
> it'd be better to panic

hard no on this

> it will never recover

wdym, it just continues with the next item.

> It constantly tries deserializing the same garbage and fails over and over

if your sub type does not match the object the node returns, yes, but imo this is more convenient in most cases, but I agree there should be an option to get the error